flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/casting/CastRulesTest.java

```diff
@@ -639,29 +639,36 @@
                         .fromCase(DOUBLE(), 0.0d, false)
                         .fromCase(DOUBLE(), -0.12345678d, true),
                 CastTestSpecBuilder.testCastTo(BINARY(2))
-                        .fromCase(CHAR(3), fromString("foo"), new byte[] {102, 111, 111})
+                        .fromCase(CHAR(3), fromString("foo"), new byte[] {102, 111, 111}, true)
+                        .fromCase(CHAR(3), fromString("foo"), new byte[] {102, 111})
+                        .fromCase(CHAR(1), fromString("f"), new byte[] {102})
+                        .fromCase(CHAR(3), fromString("f"), new byte[] {102})
+                        .fromCase(VARCHAR(5), fromString("Flink"), new byte[] {70, 108})
```

Comment:
nit: Here and below, I would use the last arg with `false` to be more visible how the test behaves with legacy behaviour enabled and disabled.

Reply:
> nit: Here and below, I would use the last arg with `false` to be more visible how the test behaves with legacy behaviour enabled and disabled.

Good point! Thanks for your review, I will update here.